pkgs/by-name/ai/airgorah/package.nix

```
@@ -0,0 +1,68 @@
+{
+  rustPlatform,
+  lib,
+  fetchFromGitHub,
+  pkg-config,
+  glib,
+  pango,
+  gdk-pixbuf,
+  graphene,
+  gtk4,
+  copyDesktopItems,
+  makeDesktopItem,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "airgorah";
+  version = "0.7.3";
+
+  src = fetchFromGitHub {
+    owner = "martin-olivier";
+    repo = "airgorah";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-cIb40TKkk3gfy4dTP8WyZqQkRGj5nItaQ3NSfexCUOA=";
+  };
+
+  cargoHash = "sha256-sJFOpxBW6Z8ewfyhJuxZByJNgaEmezw15OQt9KjFaec=";
+
+  nativeBuildInputs = [
+    pkg-config
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    glib
+    pango
+    gdk-pixbuf
+    graphene
+    gtk4
+  ];
+
+  dessktopItems = [
+    (makeDesktopItem {
+      name = "airgorah";
+      comment = "A WiFi auditing software that can perform deauth attacks and passwords cracking";
+      desktopName = "airgorah";
+      exec = "pkexec airgorah";
+      icon = "${src}/icons/app_icon.png";
```

`src` contains the entire source code this will unnecessarily include the source code when fetching this package from the cache.

Any other store path is also not recommended, because some desktop environments like to copy these `.desktop` files to another location, that's no longer in the nix store, so the file might become unusable when the app gets an update and the old app gets GC'd.

The proper way to do this is to copy the `app_icon.png` file to `$out/share/icons/hicolor/1024x1024/apps/airgorah.png` or to `$out/share/pixmaps/airgorah.png` and specify `icon = "airgorah";`.

```suggestion
      icon = "airgorah";
```